### PR TITLE
Order import section by config

### LINKED ISSUE #74: support config the order of sections
Now the order of sections is built-in, users cannot change it.

It would be good to support it.

### PR DESCRIPTION
**Pull request by vorotislav**
Closes #74 

added the ability to change the order of processing of import sections by config

### CODE REVIEW

**Comment by daixiang0** [comment]
Please add a flag for this feature since some users do not want to config order.

File: pkg/gci/internal/testdata/order.cfg.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sections:
```
Comment (jetexe) [inline pkg/gci/internal/testdata/order.cfg.yaml:1]: Rename testcase into "default order"

File: pkg/gci/internal/testdata/order2.cfg.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+custom-order: true
```
Comment (jetexe) [inline pkg/gci/internal/testdata/order2.cfg.yaml:1]: Rename testcase into "custom order"

File: cmd/gci/gcicommand.go [hunk 3]
```diff
@@ -55,6 +56,7 @@ dot - dot section, contains all dot imports. This section is not presed unless e
 
 	skipGenerated = cmd.Flags().Bool("skip-generated", false, "Skip generated files")
 
+	customOrder = cmd.Flags().Bool("custom-order", false, "Enable custom order of section")
```
Comment (jetexe) [inline cmd/gci/gcicommand.go:59]:
```suggestion
	customOrder = cmd.Flags().Bool("custom-order", false, "Enable custom order of sections")
```

Comment (jetexe) [inline cmd/gci/gcicommand.go:59]: Add this flag into `README.md`

**Review by jetexe — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -59,7 +59,7 @@ Flags:
                           default - default section, contains all rest imports
                           blank - blank section, contains all blank imports. This section is not presed unless explicitly enabled. (default [standard,default])
       --skip-generated    Skip generated files
-
+      --custom-order      Enable custom order of sections
```
Comment (daixiang0) [inline README.md:62]: Please add default comment.

File: pkg/config/config.go [hunk 5]
```diff
@@ -13,6 +13,7 @@ type BoolConfig struct {
 	NoPrefixComments bool `yaml:"no-prefixComments"`
 	Debug            bool `yaml:"-"`
 	SkipGenerated    bool `yaml:"skipGenerated"`
+	CustomOrder      bool `yaml:"custom-order"`
```
Comment (daixiang0) [inline pkg/config/config.go:16]: nit: `yaml:"customOrder"`. 

`NoPrefixComments` would be removed in the following release.

File: pkg/gci/gci.go [hunk 6]
```diff
@@ -154,56 +154,63 @@ func LoadFormatGoFile(file io.FileObj, cfg config.Config) (src, dist []byte, err
...
-		body = append(body, utils.Linebreak)
-	}
 
-	if len(result["default"]) > 0 {
-		for _, d := range result["default"] {
-			AddIndent(&body, &firstWithIndex)
-			body = append(body, src[d.Start:d.End]...)
+	// if default order sorted sections
+	if !cfg.CustomOrder {
+		orders := map[string]int{
```
Comment (daixiang0) [inline pkg/gci/gci.go:160]: make it defined in the header of file as `defaultOrder`.

File: pkg/gci/gci.go [hunk 7]
```diff
@@ -154,56 +154,63 @@ func LoadFormatGoFile(file io.FileObj, cfg config.Config) (src, dist []byte, err
...
-		body = append(body, utils.Linebreak)
-	}
+			return orders[sectionI] < orders[sectionJ]
+		})
+	}
+
+	// order by section list
+	for _, s := range cfg.Sections {
+		if strings.HasPrefix(s.String(), "prefix(") {
+
```
Comment (daixiang0) [inline pkg/gci/gci.go:186]: remove this empty line.

Comment (vorotislav) [reply pkg/gci/gci.go:186]: done

File: pkg/gci/gci.go (hunk 6 above)
Comment (vorotislav) [reply pkg/gci/gci.go:160]: done

File: pkg/config/config.go (hunk 5 above)
Comment (vorotislav) [reply pkg/config/config.go:16]: done

File: README.md (hunk 4 above)
Comment (vorotislav) [reply README.md:62]: Please help. Is this comment enough?
`--custom-order      Enable custom order of sections. The default order is standard: standard > default > custom > blank > dot.`

Comment (daixiang0) [reply README.md:62]: `If specified, make Make the section order the same as your configuration order. The default order is standard > default > custom > blank > dot.`

File: pkg/gci/gci.go [hunk 8]
```diff
@@ -154,56 +162,54 @@ func LoadFormatGoFile(file io.FileObj, cfg config.Config) (src, dist []byte, err
...
-		body = append(body, utils.Linebreak)
-	}
 
-	if len(result["default"]) > 0 {
-		for _, d := range result["default"] {
-			AddIndent(&body, &firstWithIndex)
-			body = append(body, src[d.Start:d.End]...)
-		}
+	// if default order sorted sections
+	if !cfg.CustomOrder {
```
Comment (daixiang0) [inline pkg/gci/gci.go:167]: I notice that we will sort it for each file which is unnecessary. Please sort it when loading config.

File: README.md (hunk 4 above)
Comment (vorotislav) [reply README.md]: Thanks. Done

File: pkg/gci/gci.go (hunk 8 above)
Comment (vorotislav) [reply pkg/gci/gci.go]: Agree with you. Done

**Pull request merged**